tornado/test/httpclient_test.py

```diff
@@ -415,6 +426,18 @@ def test_gzip(self):
         f = gzip.GzipFile(mode="r", fileobj=response.buffer)
         self.assertEqual(f.read(), b"asdfqwer")
 
+    def test_invalid_gzip(self):
+        # test if client hangs on tricky invalid gzip
+        with ExpectLog(
+            app_log, "(Uncaught exception|Exception in callback)", required=False
```

Comment:
Why isn't this message required? (and the HTTPError below?) Shouldn't the bad gzip data consistently return an error?

Reply:
In many cases the extra data after the gzip data is silently ignored. It seems to vary depending on libcurl version, and python version, e.g. pypy never has a problem? I considered trying a whole range of sizes to see if different versions have different edge-case sizes depending on some internal buffer sizes somewhere ...

Anyway I'll temporarily change this to required=True and see exactly which cases don't do this.